src/topology/metric_space/basic.lean

```diff
@@ -456,6 +462,15 @@ show dist x x ≤ ε, by rw dist_self; assumption
 @[simp] lemma closed_ball_eq_empty : closed_ball x ε = ∅ ↔ ε < 0 :=
 by rw [← not_nonempty_iff_eq_empty, nonempty_closed_ball, not_le]
 
+/-- Closed balls and spheres coincide only when the radius is non-positive -/
+@[simp] theorem closed_ball_eq_sphere_of_nonpos (hε : ε ≤ 0) : closed_ball x ε = sphere x ε :=
```

Comment:
I'm not sure that this should be a `simp` lemma.

Reply:
I think it's pretty harmless; do you feel strongly? `sphere` is often nicer than `closed_ball` as it contains an equality.

Reply:
My only concern is that we seldom need closed balls or spheres of nonpositive radius, and making `simp` look for `r ≤ 0` every time it sees `closed_ball x r` looks like a waste of time. OTOH, I don't know how much time does it take to fail.

Reply:
Fair enough, I'll remove it.